bokeh/models/mappers.py

```diff
@@ -16,6 +16,17 @@ class ColorMapper(Model):
 
     """
 
+    palette = Seq(Color, help="""
+    A sequence of colors to use as the target palette for mapping.
+
+    This property can also be set as a ``String``, to the name of
+    any of the palettes shown in :ref:`bokeh.palettes`.
+    """).accepts(Enum(Palette), lambda pal: getattr(palettes, pal))
+
+    def __init__(self, palette=None, **kwargs):
+        if palette is not None: kwargs['palette'] = palette
+        super(ColorMapper, self).__init__(**kwargs)
+
```

Comment:
I left `low` and `high` on the specific color mappers (log and linear) because I could imagine cases they don't quite work (adding a symmetric log mapper for positive/negative value log mapping).

In the future I could imagine having color functions that are iterators, but that seems a little bit out. Moving palette onto ColorMapper also removes a lot of duplicated code.